pytradfri/resource.py

```diff
@@ -5,39 +5,102 @@
 from datetime import datetime
 from typing import Any, Callable, Dict, List, Union
 
+from pydantic import BaseModel, Field
+
+from pytradfri.device.signal_repeater import SignalRepeaterResponse
+
 from .command import Command, TypeProcessResultCb
-from .const import ATTR_CREATED_AT, ATTR_ID, ATTR_NAME
-from .typing import ApiResourceResponse
+from .const import (
+    ATTR_APPLICATION_TYPE,
+    ATTR_CREATED_AT,
+    ATTR_DEVICE_BATTERY,
+    ATTR_DEVICE_FIRMWARE_VERSION,
+    ATTR_DEVICE_INFO,
+    ATTR_DEVICE_MANUFACTURER,
+    ATTR_DEVICE_MODEL_NUMBER,
+    ATTR_DEVICE_POWER_SOURCE,
+    ATTR_DEVICE_SERIAL,
+    ATTR_ID,
+    ATTR_LAST_SEEN,
+    ATTR_LIGHT_CONTROL,
+    ATTR_NAME,
+    ATTR_OTA_UPDATE_STATE,
+    ATTR_REACHABLE_STATE,
+    ATTR_START_BLINDS,
+    ATTR_SWITCH_PLUG,
+    ROOT_AIR_PURIFIER,
+    ROOT_SIGNAL_REPEATER,
+)
+from .device.air_purifier import AirPurifierResponse
+from .device.blind import BlindResponse
+from .device.light import LightResponse
+from .device.socket import SocketResponse
 
 # type alias
 TypeRaw = Dict[str, Union[str, int, List[Dict[str, Union[str, int]]]]]
 TypeRawSimple = Dict[str, Union[str, int]]
 TypeRawList = Dict[str, List[Dict[str, Union[str, int]]]]
 
 
+class DeviceInfoResponse(BaseModel):
+    """Represent the device info part of the device response."""
+
+    manufacturer: str = Field(alias=ATTR_DEVICE_MANUFACTURER)
+    model_number: str = Field(alias=ATTR_DEVICE_MODEL_NUMBER)
+    serial: str = Field(alias=ATTR_DEVICE_SERIAL)
+    firmware_version: str = Field(alias=ATTR_DEVICE_FIRMWARE_VERSION)
+    power_source: int | None = Field(alias=ATTR_DEVICE_POWER_SOURCE)
+    battery_level: int | None = Field(alias=ATTR_DEVICE_BATTERY)
+
+
+class ApiResourceResponse(BaseModel):
+    """Represent an API response."""
+
+    application_type: int | None = Field(alias=ATTR_APPLICATION_TYPE)
+    created_at: int = Field(alias=ATTR_CREATED_AT)
+    device_info: DeviceInfoResponse | None = Field(alias=ATTR_DEVICE_INFO)
+    id: int = Field(alias=ATTR_ID)
+    last_seen: int | None = Field(alias=ATTR_LAST_SEEN)
+    name: str | None = Field(alias=ATTR_NAME)
+    ota_update_state: int | None = Field(alias=ATTR_OTA_UPDATE_STATE)
+    reachable: int | None = Field(alias=ATTR_REACHABLE_STATE)
+
+    air_purifier: list[AirPurifierResponse] | None = Field(alias=ROOT_AIR_PURIFIER)
+    blind: list[BlindResponse] | None = Field(alias=ATTR_START_BLINDS)
+    light: list[LightResponse] | None = Field(alias=ATTR_LIGHT_CONTROL)
+    socket: list[SocketResponse] | None = Field(alias=ATTR_SWITCH_PLUG)
+    signal_repater: list[SignalRepeaterResponse] | None = Field(
+        alias=ROOT_SIGNAL_REPEATER
+    )
+
+
 class ApiResource:
     """Base object for resources returned from the gateway."""
 
-    def __init__(self, raw: ApiResourceResponse) -> None:
+    _model_class: type[ApiResourceResponse] = ApiResourceResponse
+    raw: ApiResourceResponse
```

Comment:
We can do it like this to allow us to wait with updating all of the resource children.
```suggestion
    _model_class: type[ResourceResponse] | None = None
    raw: TypeRaw | ResourceResponse
```